backend/controllers/AuthController.ts

```diff
@@ -19,9 +25,65 @@ export class AuthController extends BaseController {
     this.router.get('/siwf', this.getSiwf.bind(this));
     this.router.get('/account', this.getAccount.bind(this));
     this.router.post('/login', this.postLogin.bind(this));
+    this.router.get('/login/frequency-access', this.getFrequencyAccessLogin.bind(this));
+    this.router.post('/login/frequency-access/verify', this.postFrequencyAccessLogin.bind(this));
     this.router.post('/logout', validateAuthToken, this.postLogout.bind(this));
   }
 
+  private validateQueryParams(params: AccountQueryParams): string | null {
+    const { msaId, referenceId, accountId } = params;
+    if (!msaId && !referenceId && !accountId) {
+      return 'msaId, referenceId, or accountId is required';
+    }
+    return null;
+  }
+
+  public async postFrequencyAccessLogin(req: Request, res: Response) {
+    try {
+      const payload = await AccountService.getInstance().then((service) => service.verifyFrequencyAccessAuth(req.body));
+
+      console.log('payload', payload);
+      res.json(payload);
+    } catch (e) {
+      if (e instanceof HttpError) {
+        res.status(e.code).send(e.message).end();
+      } else {
+        res.status(HttpStatusCode.InternalServerError).send(e).end();
+      }
+    }
+  }
+
+  public async getFrequencyAccessLogin(req: Request, res: Response) {
+    try {
+      const HOSTNAME = 'localhost';
+      const PORT = Config.instance().port;
+      const SIWF_CALLBACK = `http://${HOSTNAME}:${PORT}/login/callback`;
+
+      const params = {
```

Comment:
The client should provide this information, but we can refactor it in the future if necessary.